code/cfile/cfile.cpp

```diff
@@ -363,13 +360,16 @@ int cfile_chdir(const char *dir)
 
 int cfile_pop_dir()
 {
-	Assert(Cfile_stack_pos);
-
-	if ( !Cfile_stack_pos )
+	if (Cfile_stack.empty())
 		return -1;
 
-	Cfile_stack_pos--;
-	return cfile_chdir(Cfile_stack[Cfile_stack_pos]);
+	int rc = cfile_chdir(Cfile_stack.back().c_str());
+
+	Assertion(rc == 0, "Failed to chdir() to previous directory (%s)!", Cfile_stack.back().c_str());
```

Comment:
Might be helpful to know the value of rc here, too.

Reply:
I thought about that, but it's actually far less useful than the directory name itself. And 99% of the time this is just going back to the working directory. So if we can't do that then we're pretty much fubar.

Reply:
I should add that the rc value is not the result of the OS chdir() command itself, which is why it isn't particular useful beyond whether it worked or not.